### PR TITLE
Option to provide a custom java binary path

### LINKED ISSUE #63: Option to provide a custom Java binary path
# Description
Provide an option to supply a custom path to a Java executable instead using the default `java` as present in PATH.

# Definition of Done
* [x] In the settings menu, there is an option to provide a custom command to invoke a Java binary
* [x] The provided binary is used when invoking EvoSuite
* [x] Hardcoded references are removed

### PR DESCRIPTION
**Pull request by martinmladenov**
# Description of changes made
Added a settings option to provide a custom Java binary instead of simply using the default `java` command.

# Why is merge request needed
The default `java` command may not be compatible with EvoSuite (e.g. if it is a too old or new version).

# Other notes
Closes #63

### CODE REVIEW

File: src/main/resources/defaults/TestGenie.properties [hunk 1]
```diff
@@ -1,8 +1,9 @@
-stoppingCondition=MAXTIME
-searchBudget=60
-initializationTimeout=120
-minimizationTimeout=60
-assertionTimeout=60
-junitCheckTimeout=60
-populationLimit=INDIVIDUALS
-population=50
+stoppingCondition = MAXTIME
```
Comment (jzelenjak) [inline src/main/resources/defaults/TestGenie.properties]: Why do they have spaces near the = sign?

**Review by jzelenjak — DISMISSED** [review]
Good job, Marti!
I have tested it and it works on my machine.
The feature will definitely be useful for our plugini

**Review by kirilvasilev16 — CHANGES_REQUESTED** [review]
When a I provide an incorrect java path, I get an exception. Can you change this to have a MessageBox appear that notifies the user that an EvoSuite exception occured?

**Comment by martinmladenov** [comment]
> When a I provide an incorrect java path, I get an exception. Can you change this to have a MessageBox appear that notifies the user that an EvoSuite exception occured?

Thank you for the comment - I made it show notifications instead.

**Review by kirilvasilev16 — APPROVED** [review]
I like the notification even more than what I suggested. Good job! It works normally on Windows ;)

**Review by lyuben-todorov — APPROVED** [review]
LGTM!

**Pull request merged**